[quant][refactor tests] Move quantization tests into subfolders

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59007 [quant][refactor tests] Move quantization tests into subfolders**
* #59000 [quant][refactor tests] Move TestModelNumerics to a separate file
* #58999 [quant][refactor tests] Rename test_numeric_suite and equalization tests
* #58963 [quant][refactor tests] split test_workflow_module into test_workflow_ops and test_workflow_module

Summary:
Create folders for each test category and move the tests.
Will follow-up with a cleanup of test_quantization.py

```
test/quantization
         - bc/
            - test_backward_compatibility.py
         - core/
            - test_quantized_kernels.py
            - test_quantized_workflow_ops.py
            - test_quantized_tensor.py
            - test_workflow_module.py
         - eager/
            - test_quantize_eager_ptq.py
            - test_quantize_eager_qat.py
            - test_fusion.py
            - test_model_numerics.py
            - test_numeric_suite_eager.py
            - test_equalize_eager.py
            - test_bias_correction_eager.py
         - fx/
           - test_quantize_fx.py
           - test_numeric_suite_fx.py
         - jit/
            - test_quantize_jit.py
            - test_fusion_passes.py      
     
```

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28718742](https://our.internmc.facebook.com/intern/diff/D28718742)

supriyar committed:
[quant][refactor tests] Move quantization tests into subfolders

Summary:
Create folders for each test category and move the tests.
Will follow-up with a cleanup of test_quantization.py

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][refactor tests] Move quantization tests into subfolders"


Summary:
Create folders for each test category and move the tests.
Will follow-up with a cleanup of test_quantization.py

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28718742).

HDCharles reviewed (approved):
looks good, although it may be a good idea to do a full test and compare to the before just to make sure nothing was lost in the shuffle

supriyar committed:
Update on "[quant][refactor tests] Move quantization tests into subfolders"


Summary:
Create folders for each test category and move the tests.
Will follow-up with a cleanup of test_quantization.py

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28718742](https://our.internmc.facebook.com/intern/diff/D28718742)

[ghstack-poisoned]

supriyar committed:
Update on "[quant][refactor tests] Move quantization tests into subfolders"


Summary:
Create folders for each test category and move the tests.
Will follow-up with a cleanup of test_quantization.py

```
test/quantization
         - bc/
            - test_backward_compatibility.py
         - core/
            - test_quantized_kernels.py
            - test_quantized_workflow_ops.py
            - test_quantized_tensor.py
            - test_workflow_module.py
         - eager/
            - test_quantize_eager_ptq.py
            - test_quantize_eager_qat.py
            - test_fusion.py
            - test_model_numerics.py
            - test_numeric_suite_eager.py
            - test_equalize_eager.py
            - test_bias_correction_eager.py
         - fx/
           - test_quantize_fx.py
           - test_numeric_suite_fx.py
         - jit/
            - test_quantize_jit.py
            - test_fusion_passes.py      
     
```

Test Plan:
python test/test_quantization.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D28718742](https://our.internmc.facebook.com/intern/diff/D28718742)

[ghstack-poisoned]

supriyar commented:
@supriyar has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28718742).
